Public/Src/Tools/SymbolDaemon/SymbolIndexer.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Collections.Generic;
+using System.Diagnostics.ContractsLight;
+using System.IO;
+using Microsoft.VisualStudio.Services.Content.Common.Tracing;
+using Microsoft.VisualStudio.Services.Symbol.Common;
+
+namespace Tool.SymbolDaemon
+{
+    /// <summary>
+    /// Straightforward implementation of ISymbolIndexer
+    /// </summary>
+    public class SymbolIndexer : ISymbolIndexer
+    {
+        private readonly SymstoreUtil m_symstoreUtil;
+
+        /// <nodoc />
+        public SymbolIndexer(IAppTraceSource tracer)
+        {
+            m_symstoreUtil = new SymstoreUtil(tracer);
+        }
+
+        /// <inheritdoc/>
+        public IEnumerable<DebugEntryData> GetDebugEntries(FileInfo file, bool calculateBlobId = false)
+        {
+            Contract.Requires(File.Exists(file.FullName));

```

Comment:
>File.Exists(file.FullName) [](start = 30, length = 26)

Is it not the case that when you're able to create `FileInfo`, the file is guaranteed to exist?

Reply:
I believe one can create `FileInfo` even if the file is not there (as long as a user has enough permissions to probe the path)

---
In reply to: [313643092](https://github.com/microsoft/BuildXL/pull/727#discussion_r313643092) [](ancestors = 313643092)